fdbserver/storageserver.actor.cpp

```diff
@@ -5140,9 +5167,10 @@ ACTOR Future<Void> getMappedKeyValuesQ(StorageServer* data, GetMappedKeyValuesRe
 			try {
 				// Map the scanned range to another list of keys and look up.
 				GetMappedKeyValuesReply _r =
-				    wait(mapKeyValues(data, getKeyValuesReply, req.mapper, &req, tenantPrefix, req.matchIndex));
+				    wait(mapKeyValues(data, getKeyValuesReply, req.mapper, &req, req.matchIndex, &remainingLimitBytes));
```

Comment:
Does this still need `remainingLimitBytes` to be a pointer? As I understood it, this was (in the older PR) passed as a pointer because it needed to include the secondary fetches and then decrease that value from the primary scans limit. However, with the current design, I don't think that's a thing any more, as the first scan uses a different limit, and then the secondary scans will keep iterating through batches until they have read up to the original limit (I think)

Reply:
you are right, it is used for[ stats and metric only now.](https://github.com/apple/foundationdb/blob/9a643286cb8415b9deae2c92359cf0da47ecf7f1/fdbserver/storageserver.actor.cpp#L5180). but we might still want to keep it?